src/build_tools.rs

```diff
@@ -248,3 +248,33 @@ pub fn safe_repr(v: &PyAny) -> Cow<str> {
         "<unprintable object>".into()
     }
 }
+
+#[derive(Debug, Clone)]
+pub(crate) enum ExtraBehavior {
+    Unset,
+    Allow,
+    Forbid,
+    Ignore,
+}
+
+impl ExtraBehavior {
+    pub fn from_schema_or_config(py: Python, schema: &PyDict, config: Option<&PyDict>) -> PyResult<Self> {
+        let extra_behavior = schema_or_config::<Option<&str>>(
```

Comment:
```suggestion
        let extra_behavior: Option<&str> = schema_or_config(
```

Reply:
also, if you're using `Literal['allow', 'forbid', 'ignore', None]`, then you need to support `py.None()` here to.

Might be better change above to

```py
ExtraBehavior = Literal['allow', 'forbid', 'ignore']
```

Reply:
When / where do I need to support that? I have tests where `None` is passed in and it works.

Reply:
I think `typed_dict_schema` is filtering out the `None`, if you pass a dict directly to `SchemaValidator`, my guess is you'll get an error.

Reply:
okay changed